app/src/main/java/org/mozilla/fenix/home/recentvisits/interactor/RecentVisitsInteractor.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.recentvisits.interactor
+
+import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem.RecentHistoryGroup
+import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem.RecentHistoryHighlight
+
+/**
+ * All possible user interactions with the "Recent visits" section.
+ */
+interface RecentVisitsInteractor {
+
+    /**
+     * User clicked on the "Show all" button besides the recent visits header.
```

Comment:
The kDoc deviates quite a bit from our established pattern for Interactor kDocs. I would prefer to preserve the existing pattern where the kDoc of an Interaction function should describe what the controller will do (since we usually use @See in the controller function and that extra bit of information is much more helpful when learning and navigating the code otherwise I think the current description as-is does not add much since you can just read the function name as well) and what user interaction triggers the function call.

Reply:
The same as the above, I'd say the interface is just a general contract with different implementations then having different behavior.
The controller is passed as an argument to an implementation of this interface (it might not be for all) so the general contract would not know what happens after. 
Loose coupling I'd say it's an advantage in the long term that contributes to the cleanliness of the code now.

Reply:
To reiterate https://github.com/mozilla-mobile/fenix/pull/22535#discussion_r754654883, I think we should go back towards the "Callback" phrasing of these descriptions that you used in the Pocket Interactor/Controllers. I think what you have proposed is good, but I would like to ensure we are consistent with how we write these kDocs throughout the codebase so that will serve as something that people will continue to emulate and repeat.